presto-iceberg/src/test/java/com/facebook/presto/iceberg/hadoop/TestIcebergDistributedHadoop.java

```diff
@@ -26,4 +41,35 @@ public TestIcebergDistributedHadoop()
     {
         super(HADOOP);
     }
+
+    @Override
+    protected QueryRunner createQueryRunner()
```

Comment:
I noticed that there're quite a number of similar `createQueryRunner()` implementations among the `hadoop` and `nessie` testing folders. Is it possible to refactor them into less duplicate code?

Reply:
Totally agree about the duplication being less-than-ideal. The issue I was running into is that both Hive and native catalogs pull a property value (`hive.metastore.uri` for Hive and `iceberg.catalog.warehouse` for native) from an already instantiated `QueryRunner`, but the `IcebergQueryRunner` class is static method-only. I could add a new static method to the `IcebergQueryRunner` that's specific to native catalogs if we think that would be the better approach. And I'm certainly open to other suggestions that are slipping my mind as well!

Reply:
As you have also noticed, for now, take the `hadoop` catalog as an example, both `TestIcebergDistributedHadoop` and `TestIcebergSmokeHadoop` have `createQueryRunner` functions and these functions are exactly the same. Moreover, the `createQueryRunner` functions for testing Hive/Hadoop/Nessie catalogs are pretty similar, only with some property differences.

Reply:
The method you mentioned about adding a new static method (or more) looks reasonable to me in able to reduce the duplicate code. Just a rough idea for your reference: we may also create a factory class to assemble necessary properties based on the catalog type we want to test. This class can return a corresponding `QueryRunner` for testing.

Reply:
No need to solve it completely in this PR, but a follow-up PR on this refactoring will be much appreciated.

Reply:
Great point about the factory method - thanks for the idea! I'll create a follow up PR early next week